zproject/backends.py

```diff
@@ -414,7 +414,7 @@ class ZulipLDAPAuthBackend(ZulipLDAPAuthBackendBase):
     def authenticate(self, username, password, realm_subdomain=None, return_data=None):
         # type: (Text, str, Optional[Text], Optional[Dict[str, Any]]) -> Optional[UserProfile]
         try:
-            if settings.REALMS_HAVE_SUBDOMAINS:
+            if settings.REALMS_HAVE_SUBDOMAINS and realm_subdomain is not None:
```

Comment:
Ah, I merged this hunk to master but was too hasty in doing so -- it actually breaks a test, like so:
https://travis-ci.org/zulip/zulip/jobs/262921012
So I've reverted that. This looked right semantically, but clearly there's more investigation needed.

Your original commit may disappear when you rebase, in which case you might cherry-pick to get it back.